CharlieBackend.Api.UnitTes/ScheduleServiceTests.cs

```diff
@@ -0,0 +1,185 @@
+﻿using AutoMapper;
+using CharlieBackend.Business.Services;
+using CharlieBackend.Core.DTO.Schedule;
+using CharlieBackend.Core.Entities;
+using CharlieBackend.Core.Mapping;
+using CharlieBackend.Core.Models.ResultModel;
+using CharlieBackend.Data.Repositories.Impl.Interfaces;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+using CharlieBackend.Business.Services.ScheduleServiceFolder;
+using FluentAssertions;
+using CharlieBackend.Business.Services.Interfaces;
+
+namespace CharlieBackend.Api.UnitTest
+{
+    public class ScheduleServiceTests : TestBase
+    {
+        private readonly IMapper _mapper;
+        private readonly IScheduledEventHandlerFactory _scheduledEventFactory;
+
+        public ScheduleServiceTests()
+        {
+            _mapper = GetMapper(new ModelMappingProfile());
+            _scheduledEventFactory = new ScheduledEventHandlerFactory();
+        }
+
+        private void Initialize(CreateScheduleDto createScheduleDto, Mock<IThemeRepository> themeRepositoryMock = null,
+            Mock<IMentorRepository> mentorRepositoryMock = null, Mock<IStudentGroupRepository> studentGroupRepositoryMock = null)
+        {
+            var scheduleRepositoryMock = new Mock<IScheduledEventRepository>();
+
+            scheduleRepositoryMock.Setup(x => x.AddRange(new List<ScheduledEvent>()
+            {
+                new ScheduledEvent
+                {
+                    ThemeId = 5,
+                    StudentGroupId = 3,
+                    MentorId = 1
+                }
+            }
+            ));
+
+            var eventOccuranceRepositoryMock = new Mock<IEventOccurrenceRepository>();
+
+            eventOccuranceRepositoryMock.Setup(x => x.Add(
+                new EventOccurrence
+                {
+                    StudentGroupId = 3,
+                    EventStart = createScheduleDto.Range.StartDate,
+                    Pattern = createScheduleDto.Pattern.Type
+                }
+                ));
+
+            _unitOfWorkMock.Setup(x => x.ScheduledEventRepository).Returns(scheduleRepositoryMock.Object);
```

Comment:
Please try to make create ScheduleRepositoryMock as a field. Then in the constructor make this setup.
And then in each test configure mocked object that is stored in the field.

Reply:
Ok, done